File: 백준/Silver/1699. 제곱수의 합/제곱수의 합.cc
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    int n;
    cin >> n;
    
    vector<int> dp(n + 1, n + 1);
    dp[0] = 0;
    
    for(int i = 1; i <= n; i++){
        for(int d = 1; d * d <= i; d++){
            dp[i] = min(dp[i], dp[i - d * d] + 1);
        }
    }
    
    cout << dp[n];

    return 0;
}

File: 백준/Silver/1699. 제곱수의 합/제곱수의 합.py
import sys
import math

n = int(sys.stdin.readline())
dp = [n + 1] * (n + 1)
dp[0] = 0
for i in range(1, n + 1):
    for d in range(1, int(math.sqrt(i)) + 1):
        dp[i] = min(dp[i], dp[i - d * d] + 1)

print(dp[n])